misc.c

```diff
@@ -969,3 +969,80 @@ dpi_initialize(options_t *options)
 
     DpiSetScale(options, dpix);
 }
+
+#define PLAP_CLASSID "{4fbb8b67-cf02-4982-a7a8-3dd06a2c2ebd}"
+int
+GetPLAPRegistrationStatus(void)
+{
+    int res = 0;
+    HKEY regkey;
+    wchar_t dllPath[MAX_PATH];
+
+    _sntprintf_0(dllPath, L"%ls%ls", o.install_path, L"bin\\libopenvpn_plap.dll");
+    if (!CheckFileAccess(dllPath, GENERIC_READ)) {
+        res = -1;
+    }
+    else if (RegOpenKeyExW(HKEY_CLASSES_ROOT, L"CLSID\\"PLAP_CLASSID, 0, KEY_READ, &regkey)
+              == ERROR_SUCCESS) {
+        res = 1;
+        RegCloseKey(regkey);
+    }
+
+    return res;
+}
+
+DWORD
+SetPLAPRegistration(BOOL value)
+{
+    DWORD res = -1;
+    const wchar_t *cmd = L"C:\\windows\\system32\\reg.exe";
+    wchar_t params[MAX_PATH];
+
+    /* Run only if the state has changed */
+    int plap_status = GetPLAPRegistrationStatus();
+    if (plap_status > 0 && (BOOL) plap_status == value)  return 0;
+
+    if (value) {
```

Comment:
There is an assumption here that if plap dll is present, then .reg files are present too. This is probably fine since we expect installer to copy those 3 files together.

Reply:
That is the assumption.
Same with checking the registry entries to determine already registered -- not all entries required for COM registration are checked.